frontend/catalyst/jit.py

```diff
@@ -577,9 +577,11 @@ def capture(self, args):
         dynamic_sig = get_abstract_signature(dynamic_args)
         full_sig = merge_static_args(dynamic_sig, args, static_argnums)
 
-        def closure(*args, **kwargs):
-            st_argnums = kwargs.pop("static_argnums", static_argnums)
-            return QFunc.__call__(*args, static_argnums=st_argnums, **kwargs)
+        def closure(qnode, *args, **kwargs):
+            params = {}
+            params["static_argnums"] = kwargs.pop("static_argnums", static_argnums)
+            params["out_tree_expected"] = kwargs.pop("out_tree_expected", [])
+            return QFunc.__call__(qnode, *args, **dict(params, **kwargs))
```

Comment:
Doesn't the last line just put the params back into kwargs? 😅 Maybe faster to just modify `kwargs` directly?

Reply:
Now that I'm thinking about it, why do we pop `out_tree_expected` from the QNode call arguments that the user puts in there, do we expect and allow for this?

Reply:
`static_argnums` at least is a user-facing parameter, which is typically set on a qjit function (not 100% sure why we allow it to be set in the qnode call either 🤔).

Unless I'm missing something, in addition to making sure the parameter isn't already supplied by the user (as unlikely as that is) and raising an error if it is, we could also minimize those chances by indicating it is "private" with `_out_tree_expected`.

Reply:
> Doesn't the last line just put the params back into kwargs? 😅 Maybe faster to just modify `kwargs` directly?

So this is little hack to make the `out_tree_expected` be modified. If I just use the 
`**kwargs` it will be passed as a keword argument and if edited in the qfunc __call__ method, it is not passed above to the caller.

Reply:
> Now that I'm thinking about it, why do we pop `out_tree_expected` from the QNode call arguments that the user puts in there, do we expect and allow for this?

you are right. no need to pop it. I just need to define it here as an empty list.

Reply:
> `static_argnums` at least is a user-facing parameter, which is typically set on a qjit function (not 100% sure why we allow it to be set in the qnode call either 🤔).
> 
> Unless I'm missing something, in addition to making sure the parameter isn't already supplied by the user (as unlikely as that is) and raising an error if it is, we could also minimize those chances by indicating it is "private" with `_out_tree_expected`.

I changed it to _out_tree_expected. for static_argnums, it is set there to make sure nested qnodes work with static_argnums. I don't necessarily need it to be part of the params, but I thought it would be cleaner. Though I am fine leaving it the it was if you think it causes confusion since one is user defined and the other is not.

Reply:
Sounds good 👍